scala/private/dependency.bzl

```diff
@@ -6,12 +6,13 @@ def new_dependency_info(
         dependency_mode,
         unused_deps_mode,
         strict_deps_mode,
+        compiler_deps_mode,
         dependency_tracking_method):
     is_strict_deps_on = strict_deps_mode != "off"
     is_unused_deps_on = unused_deps_mode != "off"
 
     need_direct_jars = is_strict_deps_on or is_unused_deps_on
-    need_direct_targets = is_unused_deps_on
+    need_direct_targets = is_unused_deps_on or is_strict_deps_on
```

Comment:
Looks like both `need_direct_jars` and `need_direct_targets` will always have same value. Maybe these needs to be squashed into single flag if it's needed

Reply:
thanks for catching - simplified the code